tough/src/lib.rs

```diff
@@ -266,28 +271,212 @@ impl<'a, T: Transport> Repository<'a, T> {
         //   HASH is one of the hashes of the targets file listed in the targets metadata file
         //   found earlier in step 4. In either case, the client MUST write the file to
         //   non-volatile storage as FILENAME.EXT.
-        let targets = &self.targets.signed.targets;
-        Ok(if let Some(target) = targets.get(name) {
-            let sha256 = &target.hashes.sha256.clone().into_vec();
-            let file = if self.consistent_snapshot {
-                format!("{}.{}", hex::encode(sha256), name)
+        Ok(
+            if let Some(target) = self.targets.signed.targets.get(name) {
+                let (sha256, file) = self.target_digest_and_filename(target, name);
+                Some(self.fetch_target(target, &sha256, file.as_str())?)
             } else {
-                name.to_owned()
-            };
+                None
+            },
+        )
+    }
+
+    /// Cache an entire or partial repository to disk, including all required metadata.
+    /// The cached repo will be local, using filesystem paths.
+    ///
+    /// * `metadata_outdir` is the directory where cached metadata files will be saved.
+    /// * `targets_outdir` is the directory where cached targets files will be saved.
+    /// * `targets_subset` is the list of targets to include in the cached repo. If no subset is
+    /// specified (`None`), then *all* targets are included in the cache.
+    /// * `cache_root_chain` specifies whether or not we will cache all versions of `root.json`.
+    pub fn cache<P1, P2>(
+        &self,
+        metadata_outdir: P1,
+        targets_outdir: P2,
+        targets_subset: Option<&Vec<String>>,
```

Comment:
```suggestion
        targets_subset: Option<&[S]>,
```

... `where S: AsRef<str>`, to make the set of types that can be used here much wider (you can pass a reference to a `Vec<String>`, or a `&[&str]`, or anything in between).

Reply:
Nice, tip.